### PR TITLE
Add button to duplicate script

### LINKED ISSUE #4677: Automation/Script Editors: Ability to duplicate
## The request

Add the ability to duplicate an automation/script from the editor.

## The alternatives

Have to go to your automations/scripts.yaml file and copy/paste and then manually update the id, then reload automations/scripts.

### PR DESCRIPTION
**Pull request by cnorick**
## Proposed change

Now you can duplicate scripts just like you do for automations.
demo

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #4677
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by cnorick** [comment]
I added the new translation strings to en.json. Is that all I have to do for i18n or are there more steps I missed?

File: src/panels/config/script/ha-script-editor.ts [hunk 1]
```diff
@@ -550,6 +573,30 @@ export class HaScriptEditor extends KeyboardShortcutMixin(LitElement) {
...
+      ) {
+        return;
+      }
+      // Wait for dialog to complate closing
+      await new Promise((resolve) => setTimeout(resolve, 0));
+    }
+    showScriptEditor(this, {
+      ...this._config,
+      alias: `${this._config?.alias} (${this.hass.localize(
+        "ui.panel.config.script.picker.duplicate"
```
Comment (bramkragten) [inline src/panels/config/script/ha-script-editor.ts:595]: We should share this variable with the automation editor

Comment (cnorick) [reply src/panels/config/script/ha-script-editor.ts:595]: You mean share the alias?

Comment (bramkragten) [reply src/panels/config/script/ha-script-editor.ts:595]: use: `[%key:ui::panel::config::automation::picker::duplicate%]` in `en.json`

Comment (cnorick) [reply src/panels/config/script/ha-script-editor.ts]: Ah, I see. done.

File: src/panels/config/script/ha-script-editor.ts [hunk 2]
```diff
@@ -550,6 +573,30 @@ export class HaScriptEditor extends KeyboardShortcutMixin(LitElement) {
...
+      ) {
+        return;
+      }
+      // Wait for dialog to complate closing
+      await new Promise((resolve) => setTimeout(resolve, 0));
+    }
+    showScriptEditor(this, {
+      ...this._config,
+      alias: `${this._config?.alias} (${this.hass.localize(
+        "ui.panel.config.automation.picker.duplicate"
```
Comment (bramkragten) [inline src/panels/config/script/ha-script-editor.ts:595]: This should still be script, but in the `en.json` you can reference the translation of the automation section.
```suggestion
        "ui.panel.config.script.picker.duplicate"
```

Comment (cnorick) [reply src/panels/config/script/ha-script-editor.ts:595]: Oh, that's cool! Let me try again.

**Review by bramkragten — APPROVED** [review]

File: src/panels/config/script/ha-script-editor.ts (hunk 2 above)
Comment (cnorick) [reply src/panels/config/script/ha-script-editor.ts]: Does that look right?

**Comment by cnorick** [comment]
Just wanted to follow up with this PR. Is there anything else I need to do to get this one merged? @bramkragten

**Comment by bramkragten** [comment]
Nope!
